dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/domain/event/LegacyToSpanEventMapper.kt

```diff
@@ -83,8 +86,8 @@ internal class LegacyToSpanEventMapper(
         )
     }
 
-    private fun Long?.toStringOrNull(): String? {
-        return if (this != null && this > 0) this.toString() else null
+    private fun Long?.stringOrNullIfSmaller(threshold: Int = 0): String? {
```

Comment:
I'm not really a fan of this name and this extension. First of all the strength can be negative (you can check in the logs that we send). 
The goal of an extension is to make the code more readable, but here, especially when you're using the default value, it adds a question mark . What do `networkInfo.downKbps.stringOrNullIfSmaller()`  returns ? Smaller than what ? 

I'd rather have a `toStringIfDifferentThan(defaultValue : Long)` (and let's not make the parameter have a default value).

Reply:
The part where the `strength` is negative was already solved in the previous PR. I just need to rebase this one.